test/cached-build.test.js

```diff
@@ -74,10 +75,20 @@ test('it should correctly generate all required pino files when the cache is ena
         )
       )
 
+      execa(process.argv[0], [resolve(distFolder, firstFile)]).then(({ stdout }) => {
+        t.match(stdout, /This is first!/)
+      })
+
+      const secondDistFilePath = resolve(distFolder, `abc/cde/${secondFile}`)
+
       // Check that the root file starts with the banner and has the right path to pino-file
-      const secondContent = readFileSync(resolve(distFolder, `abc/cde/${secondFile}`), 'utf-8')
+      const secondContent = readFileSync(secondDistFilePath, 'utf-8')
       t.ok(secondContent.startsWith(banner))
       t.ok(secondContent.includes(`'pino-pretty': pinoWebpackAbsolutePath('../../${pinoPretty}')`))
+
+      execa(process.argv[0], [secondDistFilePath]).then(({ stdout }) => {
+        t.match(stdout, /This is second!/)
+      })
```

Comment:
let's use async/await for a more modern look of this code :) here and elsewhere please

Reply:
https://github.com/pinojs/pino-webpack-plugin/pull/20#discussion_r775447378

With async/await the error management would become different.

Reply:
There is no error management you're doing

Reply:
Not mine but webpack's

I tried and in both cases when I throw an error inside the `onDone` callback the result would be an unhandledRejection so no difference here.

Anyway I think that "don't mix promises with callbacks" is a more general rule useful to avoid unintended and difficult to predict behavoirs.

/cc @mcollina